src/concourse/pipelines/infrastructure/codejail/packer_pulumi_pipeline.py

```diff
@@ -0,0 +1,81 @@
+from concourse.lib.constants import PULUMI_CODE_PATH
+from concourse.lib.jobs.infrastructure import packer_jobs, pulumi_jobs_chain
+from concourse.lib.models.fragment import PipelineFragment
+from concourse.lib.models.pipeline import GetStep, Identifier, Pipeline
+from concourse.lib.resources import git_repo, registry_image
+
+codejail_container_resource = registry_image(
+    name=Identifier("codejail-container"),
+    image_repository="mitodl/codejail",
+)
+
+codejail_image_code = git_repo(
+    Identifier("ol-infrastructure-packer"),
+    uri="https://github.com/mitodl/ol-infrastructure",
+    paths=[
+        "src/bilder/components/",
+        "src/bilder/images/codejail/",
+    ],
+)
+
+codejail_pulumi_code = git_repo(
+    name=Identifier("ol-infrastructure-pulumi"),
+    uri="https://github.com/mitodl/ol-infrastructure",
+    paths=[
+        "src/ol_infrastructure/applications/codejail/",
+        "pipelines/infrastructure/scripts/",
+    ],
+)
+
+container_dependencies = [GetStep(get=codejail_container_resource.name, trigger=True)]
+
+codejail_ami_fragment = packer_jobs(
+    dependencies=container_dependencies,
+    image_code=codejail_image_code,
+    packer_template_path="src/bilder/images/codejail/codejail.pkr.hcl",
+    env_vars_from_files={"CODEJAIL_VERSION": "codejail-container/tag"},
+)
+
+for network in ["mitx", "mitx-staging", "mitxonline"]:
+    codejail_pulumi_fragment = pulumi_jobs_chain(
+        codejail_pulumi_code,
+        stack_names=[
+            f"applications.codejail.{network}.{stage}"
+            for stage in ("QA", "Production", "CI")
```

Comment:
```suggestion
            for stage in ("CI", "QA", "Production")
```
The stages need to be sequenced in the order that they will be executed. Also, `mitxonline` shouldn't have a `CI` stage because it is only deployed to a `QA` and `Production` environment.